### PR TITLE
Shell sort

### PR DESCRIPTION
**Pull request by underhood31**
Added new sorting algorithm: shell-sort.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -107,4 +107,4 @@ This project is licensed under the MIT License - see the [LICENSE](LICENSE) file
 
 #### Argument Parsing
 
-- [argparse](https://docs.python.org/3.6/library/argparse.html)
\ No newline at end of file
+- [argparse](https://docs.python.org/3.6/library/argparse.html)
```
Comment (becurrie) [inline README.md:110]: What changed here exactly?

File: py_sorter/sorter.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ def parse_args(args):
     sorting_group = parser.add_mutually_exclusive_group(required=True)
     sorting_group.add_argument('-s', '--sort', type=str,
                                choices=['bubble', 'bogo', 'merge', 'selection', 'quick', 'radix', 'insertion',
-                                        'insertion_recursive', 'heap'],
+                                        'insertion_recursive', 'heap','shell'],
```
Comment (becurrie) [inline py_sorter/sorter.py]: Use consistent format in regards to any commas in lists.

```suggestion
                                        'insertion_recursive', 'heap', 'shell'],
```

File: py_sorter/sorter.py [hunk 3]
```diff
@@ -130,6 +130,8 @@ def sort(args):
         sorted_list = selection_sort(original_list)
     elif args.sort == 'merge':
         sorted_list = merge_sort(original_list)
+    elif args.sort=='shell':
```
Comment (becurrie) [inline py_sorter/sorter.py]:
```suggestion
    elif args.sort == 'shell':
```

File: py_sorter/sorter.py [hunk 4]
```diff
@@ -130,6 +130,8 @@ def sort(args):
...
+        sorted_list=shell_sort(original_list)
```
Comment (becurrie) [inline py_sorter/sorter.py]:
```suggestion
        sorted_list = shell_sort(original_list)
```

File: py_sorter/sorts.py [hunk 5]
```diff
@@ -291,3 +291,32 @@ def max_heapify(arr, i, n):
     if largest != i:
         arr[i], arr[largest] = arr[largest], arr[i]
         max_heapify(arr, largest, n)
+
+def shell_sort(integers):
+    """The idea of shellSort is to allow exchange of far items. 
+    An h-sorted array is made for a large value of h. 
+    We keep reducing the value of h until it becomes 1. """
```
Comment (becurrie) [inline py_sorter/sorts.py]: end of docstrings should be underneath the explanation.

```suggestion
    We keep reducing the value of h until it becomes 1.
    """
```

**Review by becurrie — CHANGES_REQUESTED** [review]
Hey! Great addition of a new sort method, just some tiny syntactical changes and I'd be happy to merge! 👍

File: README.md (hunk 1 above)
Comment (underhood31) [reply README.md:110]: Nothing was intended here. Maybe some white space.

**Review by becurrie — APPROVED** [review]
Changes look good, approving 👍

**Pull request merged**